### PR TITLE
refactor(ga-screenshots): option to test single component

### PR DESCRIPTION
**Pull request by thetaPC**
Issue number: N/A

---------

## What is the current behavior?

All the E2E tests are run when performing the "Update Reference Screenshots" GitHub actions.

Screenshot 2024-01-25 at 1 31 15 PM

## What is the new behavior?

- All E2E tests still run as default.
- The GH action has the option to run a specific component instead of all tests.
  - A specified component has the option to run all tests that include the component's name in the file path
  - A specified component has the option to run all tests that are in the component's folder
- If a single component is selected, then the default is to only run tests inside it's folder.

Screenshot 2024-01-25 at 1 30 34 PM

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

## Other information

How to test:

1. Create a new branch based off this one
2. Make a style change
3. Push it to the new branch
4. Update the reference images using the new branch
5. This will show you all the options

Recommended to test:
- a component with the ion prefix (`ion-button`)
- a component without the prefix (`button`)
- a component that doesn't exist (`random-component`)
- no provided component, leave the first input blank
- the checkbox

### CODE REVIEW

**Comment by sean-perkins** [comment]
I think this might need a little more testing and exploration. Here's an example run I did on a fork of this branch: https://github.com/ionic-team/ionic-framework/actions/runs/5943476177/job/16118770162

I modified two components (button and radio) and ran this job against the branch, filtering on button.

Let me know if you have any questions!

**Comment by thetaPC** [comment]
> I think this might need a little more testing and exploration. Here's an example run I did on a fork of this branch: https://github.com/ionic-team/ionic-framework/actions/runs/5943476177/job/16118770162
> 
> I modified two components (button and radio) and ran this job against the branch, filtering on button.
> 
> Let me know if you have any questions!

Got it! I'll look into this more. Thank you!

**Comment by sean-perkins** [comment]
I'm going to remove myself from review. Feel free to re-add me once it is ready for another look!

File: .github/workflows/actions/test-core-screenshot/action.yml [hunk 1]
```diff
@@ -7,6 +7,11 @@ inputs:
     description: 'Playwright total number of test shards (ex: 4)'
   update:
     description: 'Whether or not to update the reference snapshots'
+  component:
+    description: 'The component to update the reference snapshots'
+  component-folder:
+    description: 'The component folder to update the reference snapshots'
```
Comment (sean-perkins) [inline .github/workflows/actions/test-core-screenshot/action.yml:13]: This description does not align with it's type (`boolean`). Should this match the description from `update-screenshots.yml`?

Comment (thetaPC) [reply .github/workflows/actions/test-core-screenshot/action.yml]: Hopefully, the updated description is better.

**Review by liamdebeasi — APPROVED** [review]
Works great! https://github.com/ionic-team/ionic-framework/actions/runs/7669991533

**Pull request merged**